src/Symfony/Component/DependencyInjection/Compiler/RandomizePrivateServiceIdentifiers.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Alias;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Randomizes private service identifiers, effectively making them unaccessable from outside.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class RandomizePrivateServiceIdentifiers implements CompilerPassInterface
```

Comment:
What you call a *normal* container doesn't have private services. The container shouldn't care about private services, right, but the PhpDumper *should* (and already does). So I still think that the PhpDumper should do this.